testcases/parser/parse_test.go

```diff
@@ -613,3 +616,57 @@ func TestParseTestCaseFile(t *testing.T) {
 	require.NotNil(t, testFile)
 	assert.Len(t, testFile.TestCases, 13)
 }
+
+func TestLoadAllSubstraitTestFiles(t *testing.T) {
+	got := substrait.GetSubstraitTestsFS()
+	filePaths, err := listFiles(got, ".")
+	require.NoError(t, err)
+	assert.GreaterOrEqual(t, len(filePaths), 107)
+
+	for _, filePath := range filePaths {
+		t.Run(filePath, func(t *testing.T) {
+			switch filePath {
+			case "tests/cases/boolean/bool_and.test":
+				t.Skip("Skipping bool_and.test")
+			case "tests/cases/datetime/extract.test":
+				// TODO deal with enum arguments in testcase
+				t.Skip("Skipping extract.test")
+			}
+
+			testFile, err := ParseTestCaseFileFromFS(got, filePath)
+			require.NoError(t, err)
+			require.NotNil(t, testFile)
+			reg, funcRegistry := functions.NewExtensionAndFunctionRegistries(&extensions.DefaultCollection)
+			for _, tc := range testFile.TestCases {
+				testGetFunctionInvocation(t, tc, &reg, funcRegistry)
+			}
+		})
+	}
+}
+
+func testGetFunctionInvocation(t *testing.T, tc *TestCase, reg *expr.ExtensionRegistry, registry functions.FunctionRegistry) {
+	switch tc.FuncType {
+	case ScalarFuncType:
+		invocation, err := tc.GetScalarFunctionInvocation(reg, registry)
+		require.NoError(t, err, "GetScalarFunctionInvocation failed with error in test case: %s", tc.CompoundFunctionName())
```

Comment:
Is there a test case or documentation somewhere that shows what CompoundFunctionName does?  All I can gather from here is that it doesn't return an error sometimes.

Reply:
It is similar to FunctionVariant.[CompoundName](https://github.com/substrait-io/substrait-go/blob/8e12e1e1ae704fb6235e803b9b6586504dc48c58/extensions/variants.go#L334-L334). It is a string with name and argTypes.

Added a test for TestCase.CompoundName. Earlier I was testing with ID()